Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/azure/serverless/AzureSparkServerlessCluster.java

```diff
@@ -517,6 +533,45 @@ public AzureHttpObservable getHttp() {
                 .defaultIfEmpty(this);
     }
 
+    @NotNull
+    private UpdateSparkResourcePool preparePatchResourcePool(int workerTargetInstanceCount) {
+        UpdateSparkResourcePool patchBody = new UpdateSparkResourcePool();
+
+        return patchBody
+                .withName(getName())
+                .withProperties(new UpdateSparkResourcePoolParameters()
+                        .withSparkResourceCollection(Arrays.asList(
+                                new UpdateSparkResourcePoolItemParameters()
+                                        .withName(SparkNodeType.SPARK_WORKER)
+                                        .withTargetInstanceCount(workerTargetInstanceCount)
+                        )));
+    }
+
+    private Observable<SparkResourcePool> patchResourcePoolRequest(int workerTargetInstanceCount) {
+        URI uri = getUri();
+
+        UpdateSparkResourcePool patchBody = preparePatchResourcePool(workerTargetInstanceCount);
+        if (master == null || worker == null) {
```

Comment:
Better to put the check before `UpdateSparkResourcePool patchBody = preparePatchResourcePool(workerTargetInstanceCount);`